lib/charms/opensearch/v0/opensearch_plugins.py

```
@@ -0,0 +1,277 @@
+        """Returns if the plugin needs an upgrade or not.
+
+        Needs upgrade must be set if an upgrade on the charm happens and plugins must
+        be updated.
+
+        Consider overriding this method only if the plugin needs an special care at upgrade.
+        """
+        current_version = self.version
+        # Now check if the format of this plugin and OpenSearch's match
+        num_points = len(self.distro.version.split("."))
+        return self.distro.version != current_version[:num_points]
+
+    @abstractmethod
+    def upgrade(self, uri: str) -> None:
+        """Runs the upgrade process in this plugin."""
+        pass
+
+    @abstractmethod
+    def is_enabled(self) -> bool:
+        """Returns True if the plugin is enabled."""
+        pass
+
+    @abstractmethod
+    def disable(self) -> bool:
+        """Disables the plugin.
+
+        Runs the configure() method with the configuration to disable the plugin.
+        """
+        pass
+
+    @abstractmethod
+    def enable(self) -> bool:
+        """Enables the plugin.
+
+        Runs the configure() method with the configuration to enable the plugin.
+        """
+        pass
+
+    def configure(
+        self,
+        opensearch_yml: Dict[str, str],
+        keystore: Dict[str, str] = {},
+    ) -> bool:
+        """Sets the plugin configuration. Returns True if restart needed."""
+        if not self._is_started():
+            return False
+        self._update_keystore_and_reload(keystore)
+        return any(
+            [
+                self.distro.config.put(self.CONFIG_YML, c, v)[c] == v
+                for c, v in opensearch_yml.items()
+            ]
+        )
+
+    def is_installed(self) -> bool:
+        """Returns True if the plugin name is present in the list of installed plugins."""
+        if not self._is_started():
+            return False
+        return self.name in self.distro.list_plugins()
+
+    def install(self, uri: str, batch=True) -> bool:
```

We don't need a boolean return here, an unsuccessful install should raise an exception